localstack/services/kms/provider.py

```diff
@@ -466,50 +677,169 @@ def import_key_material(
+            raise AlreadyExistsException(f"An alias with the name {alias_arn} already exists")
+        # KeyId can potentially hold one of multiple different types of key identifiers. get_key finds a key no
+        # matter which type of id is used.
+        request["TargetKeyId"] = store.get_key(request["TargetKeyId"]).metadata["KeyId"]
+        alias = KmsAlias(request)
+        store.aliases[alias_name] = alias
+
+    @handler("DeleteAlias", expand=False)
+    def delete_alias(self, context: RequestContext, request: DeleteAliasRequest) -> None:
+        store = self._get_store(context)
+        alias_name = request["AliasName"]
+        if alias_name not in store.aliases:
+            alias_arn = kms_alias_arn(request["AliasName"], context.account_id, context.region)
+            # AWS itself uses AliasArn instead of AliasName in this exception.
+            raise NotFoundException(f"Alias {alias_arn} is not found")
+        store.aliases.pop(alias_name, None)
+
+    @handler("UpdateAlias", expand=False)
+    def update_alias(self, context: RequestContext, request: UpdateAliasRequest) -> None:
+        alias_name = request["AliasName"]
+        # This API, per AWS docs, accepts only names, not ARNs.
+        validate_alias_name(alias_name)
+        store = self._get_store(context)
+        alias = store.get_alias(alias_name, context.account_id, context.region)
+        key_id = request["TargetKeyId"]
+        # Don't care about the key, just want to validate it is there.
+        store.get_key(key_id)
+        alias.metadata["TargetKeyId"] = key_id
+        alias.update_date_of_last_update()
+
+    @handler("ListAliases")
     def list_aliases(
         self,
         context: RequestContext,
         key_id: KeyIdType = None,
         limit: LimitType = None,
         marker: MarkerType = None,
     ) -> ListAliasesResponse:
-        if key_id is None:
-            return call_moto(context)
-
-        response_aliases = PaginatedList()
-        backend = self.get_kms_backend(context)
-
-        if backend.keys.get(key_id) is None:
-            raise NotFoundException(f"Unable to find key '{key_id}'")
-
-        aliases_of_key = backend.get_all_aliases().get(key_id) or []
-
-        for alias_name in aliases_of_key:
-            response_aliases.append(
-                AliasListEntry(
-                    AliasArn=kms_alias_arn(alias_name, region_name=context.region),
-                    AliasName=alias_name,
-                    TargetKeyId=key_id,
-                )
-            )
-
-        page, nxt = response_aliases.get_page(
-            lambda a: a["AliasName"], next_token=marker, page_size=limit
+        store = self._get_store(context)
+        if key_id:
+            # KeyId can potentially hold one of multiple different types of key identifiers. get_key finds a key no
+            # matter which type of id is used.
+            key_id = store.get_key(key_id).metadata["KeyId"]
```

Comment:
nit: As this seems to be used in a few places in the provider - we could wrap this in a method `get_canonical_key_id(..)` on the store, to determine the canonical key id.